File: 0923.R
# x0=1則 x1有0.4的機會是0,0.6的機會是1 
# x0=0則 x1有0.7的機會是0,0.3的機會是1 
# 試問xn=0的機率是否會受x0影響? n是個很大的數

guess=function(n,init) {
	p0=c(0.7,0.4)
	result=0
	for(j in 1:1000){
		current=init
		for(i in 1:n){
			to_zero=runif(1)>p0[current+1]
			if(to_zero){
				current=0
			}
			else{
				current=1
			}
		}
		result=result+current
	}
	result/1000
}

File: 0923.js
var _ = require('underscore');
var prop = [0.7, 0.3, 0.4, 0.6];

// Number.prototype.next = function() {
// 	return 5;
// }

function Num(val) {
	this.val = val;
}

Num.prototype.next = function() {
	var val = this.val;

	var p0 = [0.7, 0.4];

	if (Math.random() < p0[val]) {
		return new Num(0);
	} else {
		return new Num(1);

	}
};



function getProp(init, max) {
	var n = new Num(init);
	// console.log(n.next().val);
	// var max = 10000;

	var result = [];
	for (var i = 0; i < max; i++) {
		n = n.next();
		result.push(n.val);
	}

	return _.countBy(result, function(num) {
		return num == 0 ? 'zero' : 'one';
	});

}

var a = getProp(0, 1000);
console.log(a["one"] / 1000);


function mutipleChain(init, max, channelAmount) {
	var channel = [];
	for (var i = 0; i < chainAmount; i++) {
		channel.push( getProp(init, max) );
	}
}

// console.log(Math.random());